strum_macros/src/macros/enum_iter.rs

```diff
@@ -83,12 +83,14 @@ pub fn enum_iter_inner(ast: &syn::DeriveInput) -> TokenStream {
                     #(#arms),*
                 };
 
-                self.idx += 1;
+                if self.idx < #variant_count {
+                    self.idx += 1;
+                }
                 output
             }
 
             fn size_hint(&self) -> (usize, Option<usize>) {
-                let t = #variant_count - self.idx;
+                let t = if self.idx >= #variant_count { 0 } else { #variant_count - self.idx };
```

Comment:
Looks like this is unnecessary given that the above guarantees this will never underflow.

Additionally, this property is [`FusedIterator`](https://doc.rust-lang.org/std/iter/trait.FusedIterator.html) (which is not a requirement for `Iterator`, and it's the responsibility of the caller to stop calling it). If we choose to have it, then we should also signify that we have it by implementing `FusedIterator`.